funnel/models/proposal.py

```diff
@@ -164,7 +171,71 @@ def _validate_proposal_space(self, key, value):
                 redirect.proposal = self
         return value
 
-    @state.transition(None, state.DELETED)
+    # State transitions
+    state.add_conditional_state('SCHEDULED', state.CONFIRMED, lambda proposal: proposal.session is not None)
```

Comment:
After hasgeek/coaster#165, you'll need to add `label=('scheduled', __("Confirmed & Scheduled"))`